src/app/three-examples/objects/loaders/obj-loader.directive.ts

```diff
@@ -42,9 +42,10 @@ export class ObjLoaderDirective extends AbstractModelLoader {
         if (this.texturePath !== undefined) {
           this.mtlLoader.setTexturePath(this.texturePath);
         }
-        this.mtlLoader.load(this.material, material => {
-          material.preload();
-          this.loader.setMaterials(material);
+        this.mtlLoader.load(this.material, materialCreator => {
+          materialCreator.preload();          
+          var x:any = materialCreator;// seems wrong type in js module ts. Hack (as any)
+          this.loader.setMaterials(x); 
```

Comment:
Easier: `this.loader.setMaterials(materialCreator as any)`

Reply:
Btw, I found [`MaterialCreator.getAsArray()`](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/MTLLoader.d.ts#L99). Have you tried using that one?
I also wonder why the cast you're doing worked (it did, right?). I couldn't find any prototype/array indexing magic in Three.js' source code ([MTLLoader.js](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/MTLLoader.js)). So actually, it shouldn't work.

Reply:
> Btw, I found MaterialCreator.getAsArray(). Have you tried using that one?

Unfortunately, it's compiling, but failing at runtime.
```
ERROR TypeError: this.materials.create is not a function
    at OBJLoader.parse (OBJLoader.js:702)
    at Object.onLoad (OBJLoader.js:402)
```

Reply:
@makimenko Uh, sounds like a bug then. I could file it this evening, I guess, if you push your current code.